src/contributors/MapContributor.ts

```diff
@@ -3784,37 +3757,41 @@ export class MapContributor extends Contributor {
      * @returns Either the value of the field, or the value of the first item of the list of values of the corresponding field
      */
     private getValueOrFirstArrayValueFromFeature(f: Feature, flattenedField: string) {
-        return f.properties[flattenedField] || f.properties[flattenedField + '_0'];
-    }
-
-    private getBboxsForQuery(newBbox: Array<Object>) {
-        const bboxArray: Array<string> = [];
-        const numberOfBbox = newBbox.length;
-        const lastBbox = newBbox[numberOfBbox - 1];
-        const lastCoord = lastBbox['geometry']['coordinates'][0];
-        const north = lastCoord[1][1];
-        const west = this.wrap(lastCoord[2][0], -180, 180);
-        const south = lastCoord[0][1];
-        const east = this.wrap(lastCoord[0][0], -180, 180);
-        const lastBboxWSEN = west + ',' + south + ',' + east + ',' + north;
-        const lastBboxFeature = bboxPolygon([west, south, east, north]);
-        for (let i = 0; i < numberOfBbox - 1; i++) {
-            const v = newBbox[i];
-            const coord = v['geometry']['coordinates'][0];
-            const n = coord[1][1];
-            const w = this.wrap(coord[2][0], -180, 180);
-            const s = coord[0][1];
-            const e = this.wrap(coord[0][0], -180, 180);
-            const box = w + ',' + s + ',' + e + ',' + n;
-            const bboxFeature = bboxPolygon([w, s, e, n]);
-            const isbboxInclude = booleanContains(lastBboxFeature, bboxFeature);
-            const isLastBboxInclude = booleanContains(bboxFeature, lastBboxFeature);
-            if (!isbboxInclude && !isLastBboxInclude) {
-                bboxArray.push(box.trim().toLocaleLowerCase());
-            }
-        }
-        bboxArray.push(lastBboxWSEN.trim().toLocaleLowerCase());
-        return bboxArray;
+        return f.properties[flattenedField] ?? f.properties[flattenedField + '_0'];
+    }
+
+    private getGeometriesForQuery(features: Array<helpers.Feature<helpers.Geometry>>) {
+        const geometries: Array<string> = [];
+
+        const polygonFeatures = features.map(f => {
+            if (f.properties.source === 'bbox') {
+                // Compute list of anti-meridian fixed bbox features and their string representation
+                const coord = f.geometry.coordinates[0];
+                const n = coord[1][1];
+                const w = this.wrap(coord[2][0], -180, 180);
+                const s = coord[0][1];
+                const e = this.wrap(coord[0][0], -180, 180);
+                const box = w + ',' + s + ',' + e + ',' + n;
+                return {f: bboxPolygon([w, s, e, n]), str: box.trim().toLocaleLowerCase()};
+            } else {
+                // Properly orientate features
+                if (!isClockwise(f.geometry.coordinates[0])) {
```

Comment:
could you add a comment mentionning that we don't reverse internal polygons (rings) as they are not supported